app/controllers/task_lists_controller.rb

```diff
@@ -7,22 +7,50 @@ def index
 
   def show
     @task_list = TaskList.find(params[:id])
-    @tasks = Task.where(task_list_id: @task_list.id)
+    @tasks = Task.where(task_list_id: @task_list.id).where(done: false)
   end
 
-  def new; end
+  def new
+    @task_list = TaskList.new
+  end
+
+  def create
+    @task_list = TaskList.new(task_list_params)
+
+    if @task_list.save
+      redirect_to(task_lists_path)
+    else
+      render('task_lists/new')
+    end
+  end
+
+  def edit
+    @task_list = TaskList.find(params[:id])
+  end
 
-  def create; end
+  def update
+    @task_list = TaskList.find(params[:id])
 
-  def edit; end
+    if @task_list.update(task_list_params)
+      redirect_to(task_lists_path(@task_list))
+    else
+      render('task_lists/edit')
+    end
+  end
 
-  def update; end
+  def add_task
+    @task = Task.new(task_list_id: params[:task_list_id])
+  end
 
-  def destroy; end
+  def destroy
+    @task_list = TaskList.find(params[:id])
+    @task_list.destroy
```

Comment:
What happens if @task_list.destroy fails? Maybe you should put it in an `if else` statement to handle it in case it fails.